### DISCUSSION TITLE
generate a JSON property via .addProperty method on EntitySchema instance

### DISCUSSION THREAD

**Discussion by hubbusysyuhada** [mikro-orm/mikro-orm#2877] [Q&A] [answered]
Hi, I'm currently working on a project that requires me to create a json field type from EntitySchema instance. I read [https://mikro-orm.io/docs/entity-schema](url) and [https://mikro-orm.io/docs/json-properties](url) but left confused because the options available on 3rd params in `entity.addProperty()` method doesn't have something like `is_json` or so (like the `enum` for enum field ) to define what are the properties of that json column has (like `items` in `enum`)

example: in this `entity`, I want to add 1 JSON property called `json_property` which JSON contains `name`, `city`, and `age` properties

context: I need it to query that translates into `select * from "entity" where "entity"."json_property" ->> 'city' = 'london'`

<img width="419" alt="Screen Shot 2022-03-08 at 15 21 46" src="https://user-images.githubusercontent.com/76815481/157196004-d2f55cb0-dddb-4ed8-93dc-35613d99cc2d.png">

**Comment by B4nan** [comment, marked as answer]
JSON properties don't have schema, they can hold literally anything that serializes to JSON, so there is no way to specify the shape (in fact it can be just a value, not an object or array). 

If you want to have predefined shape, you can use embeddables in object mode as they will use JSON column to store the data.

Querying should work the same for both.

**Reply by hubbusysyuhada** [reply]
thank you, appreciate it!
